Refactor plan

jat-canonical opened the pull request:
## Done

* Move the plan to the new model of subcluster and controller.
* Created submodules for corresponding juju models
* Add tests for each model.

## JIRA / Launchpad bug

[AC-3296](https://warthogs.atlassian.net/browse/AC-3296)

NOTE: Should be merged after #10 


[AC-3296]: https://warthogs.atlassian.net/browse/AC-3296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jat-canonical committed:
feat(gh): move from tfsec to trivy

jat-canonical committed:
feat!: refactor the plan according to subcluster and controller models

This commit changes the plan to more closely resemble what we deploy for
customers in terms of the juju topology. The new plan has two
submodules, one for subcluster and the other for controller. The two
models are connected to each other using cross models. The juju offers
are being exposed by each terraform module to connect the two
corresponding models.

jat-canonical committed:
fix(gh): use the new inputs for generating plan for security checks

ajanon commented on README.md:
```diff
@@ -9,53 +9,53 @@ This is a terraform plan to deploy anbox cloud using juju and terraform.
 
-This plan uses a submodule to deploy an anbox subcluster the documentation for
-it can be found [here](./modules/subcluster/README.md)
+This plan uses terroform moudules to deploy an [anbox subcluster](./modules/subcluster/README.md)
```
Small typo:
```suggestion
This plan uses terraform moudules to deploy an [anbox subcluster](./modules/subcluster/README.md)
```

ajanon commented on docs/header.md:
```diff
@@ -9,7 +9,9 @@ This is a terraform plan to deploy anbox cloud using juju and terraform.
-This plan uses a submodule to deploy an anbox subcluster the documentation for
-it can be found [here](./modules/subcluster/README.md)
-
+This plan uses terroform moudules to deploy an [anbox subcluster](./modules/subcluster/README.md)
```
See above ;)

ajanon commented on modules/controller/outputs.tf:
Should this have a copyright header?

jat-canonical replied:
I'll add this in the upcoming PRs.

jat-canonical replied:
fixed in #13

ajanon reviewed (changes requested):
Some nitpicks

jat-canonical committed:
feat: add `enable_ha` variable to deploy HA configuration

This commit adds a variable to the terraform plan which allows people to
deploy HA configuration for anbox cloud.

ajanon commented on README.md:
```diff
@@ -9,53 +9,53 @@ This is a terraform plan to deploy anbox cloud using juju and terraform.
 
-This plan uses a submodule to deploy an anbox subcluster the documentation for
-it can be found [here](./modules/subcluster/README.md)
+This plan uses terraform moudules to deploy an [anbox subcluster](./modules/subcluster/README.md)
```
Small typo:
```suggestion
This plan uses terraform modules to deploy an [anbox subcluster](./modules/subcluster/README.md)
```

jat-canonical replied:
fixed in #13  thanks

ajanon reviewed (changes requested):
Small typo (please ignore if you think it doesn't matter)

morphis reviewed (approved):
Overall LGTM, I will though have to play with it to see how we can further tune/optimize.
